### DISCUSSION TITLE
Why does requirements.lock contains project in editable mode ?

### DISCUSSION THREAD

**Discussion by charbonnierg** [astral-sh/rye#841] [Q&A] [answered]
I'm used to writing docker images for Python project in two steps:

- First I copy a requirements.txt: `COPY requirements.txt /requirements.txt`
- I install the dependencies: `RUN pip install -r requirements.txt`
- Then I copy the whole project: `COPY . /project`
- And finally I install the project with `RUN pip install --no-deps /project`

This has the nice advantage that if I update my code, layers where dependencies are installed are cached, and repeated builds are much quicker.

When using `rye`, the `requirements.lock` contains the following line:

```txt
-e file:.
```

I'd like to know why this line is present, and if it is possible to configure `rye` to not include the line in the project config (`pyproject.toml`) ?

**Comment by charbonnierg** [comment]
I did see this answer which describes a workaround: https://github.com/astral-sh/rye/discussions/239#discussioncomment-6032595, however I wanted to know whether this will be a definitive behaviour or not

**Reply by bluss** [reply, marked as answer]
W.r.t definitive: no, rye does not consider its lock file design to be finished and may well change in the future.

**Reply by bluss** [reply]
Does your project fit the definition of rye init --virtual? I.e. the project is not itself a package. That's the fastest way to avoid the `-e file:.` line, because it's just installing the package itself (as editable) and if you think you can just remove this without replacement, then virtual should work for you.

**Reply by charbonnierg** [reply]
Thanks for the answers ! I do have a Python project so using --virtual is not an option.  I’ll stick to using sed to remove the unwanted requirement line in my Dockerfile.

**Reply by helderco** [reply]
I'm on the same boat but I have a suggestion:
- https://github.com/astral-sh/rye/issues/895
